kimchi/src/circuits/lookup/index.rs

```diff
@@ -217,21 +215,64 @@ impl<F: PrimeField + SquareRootField> LookupConstraintSystem<F> {
                 // product is 1, we cannot use those rows to store any values.
                 let max_num_entries = d1_size - zk_rows - 1;
 
-                //~ 2. Get the lookup selectors and lookup tables (TODO: how?)
+                //~ 2. Get the lookup selectors and lookup tables that are specified implicitly
+                // by the lookup gates (TODO: how?)
                 let (lookup_selectors, gate_lookup_tables) =
                     lookup_info.selector_polynomials_and_tables(domain, gates);
 
-                //~ 3. Concatenate runtime lookup tables with the ones used by gates
-                let mut lookup_tables: Vec<_> = gate_lookup_tables
+                // Checks whether an iterator contains any duplicates, and if yes, raises
+                // a corresponding LookupTableIdCollision error.
+                fn check_id_duplicates<'a, I: Iterator<Item = &'a i32>>(
```

Comment:
Nit: as it is strictly relatd to tables, you can move this in `tables/mod.rs`

Reply:
Since we don't use this function anywhere except `create` and it's somewhat tailored for it (works with `LookupError` that is defined in `index.rs`), I'd say it's an OK place for it? Otherwise it'd complicate dependencies and would be not obvious where it's used/why it's defined publicly.